src/main/java/org/spongepowered/api/util/TemporalUnits.java

```diff
@@ -0,0 +1,146 @@
+import java.time.temporal.TemporalUnit;
+
+/**
+ * An enumeration of all the default and minecraft related {@link TemporalUnit}s.
+ */
+public final class TemporalUnits {
+
+    // The default temporal units, from ChronoUnit
+
+    /**
+     * @see ChronoUnit#NANOS
+     */
+    public static final ChronoUnit NANOS = ChronoUnit.NANOS;
+
+    /**
+     * @see ChronoUnit#MILLIS
+     */
+    public static final ChronoUnit MILLIS = ChronoUnit.MILLIS;
+
+    /**
+     * @see ChronoUnit#SECONDS
+     */
+    public static final ChronoUnit SECONDS = ChronoUnit.SECONDS;
+
+    /**
+     * @see ChronoUnit#MINUTES
+     */
+    public static final ChronoUnit MINUTES = ChronoUnit.MINUTES;
+
+    /**
+     * @see ChronoUnit#HOURS
+     */
+    public static final ChronoUnit HOURS = ChronoUnit.HOURS;
+
+    /**
+     * @see ChronoUnit#HALF_DAYS
+     */
+    public static final ChronoUnit HALF_DAYS = ChronoUnit.HALF_DAYS;
+
+    /**
+     * @see ChronoUnit#DAYS
+     */
+    public static final ChronoUnit DAYS = ChronoUnit.DAYS;
+
+    /**
+     * @see ChronoUnit#WEEKS
+     */
+    public static final ChronoUnit WEEKS = ChronoUnit.WEEKS;
+
+    /**
+     * @see ChronoUnit#MONTHS
+     */
+    public static final ChronoUnit MONTHS = ChronoUnit.MONTHS;
+
+    /**
+     * @see ChronoUnit#YEARS
+     */
+    public static final ChronoUnit YEARS = ChronoUnit.YEARS;
+
+    /**
+     * @see ChronoUnit#DECADES
+     */
+    public static final ChronoUnit DECADES = ChronoUnit.DECADES;
+
+    /**
+     * @see ChronoUnit#CENTURIES
+     */
+    public static final ChronoUnit CENTURIES = ChronoUnit.CENTURIES;
+
+    /**
+     * @see ChronoUnit#MILLENNIA
+     */
+    public static final ChronoUnit MILLENNIA = ChronoUnit.MILLENNIA;
+
+    /**
+     * @see ChronoUnit#ERAS
+     */
+    public static final ChronoUnit ERAS = ChronoUnit.ERAS;
+
+    /**
+     * @see ChronoUnit#FOREVER
+     */
+    public static final ChronoUnit FOREVER = ChronoUnit.FOREVER;
+
+    // Minecraft related temporal units
+
+    // TODO: Move impl to sponge common?
+
+    /**
+     * A {@link MinecraftTemporalUnit} which represents one minecraft tick.
+     */
+    public static final MinecraftTemporalUnit MINECRAFT_TICKS =
+            new MinecraftTemporalUnit("MinecraftTicks", Duration.ofMillis(50));
+
+    /**
+     * A {@link MinecraftTemporalUnit} which represents a minecraft day.
+     */
+    public static final MinecraftTemporalUnit MINECRAFT_DAYS =
+            new MinecraftTemporalUnit("MinecraftDays", MINECRAFT_TICKS.getDuration().multipliedBy(24000));
+
+    /**
+     * A {@link MinecraftTemporalUnit} which represents a half minecraft day.
+     */
+    public static final MinecraftTemporalUnit HALF_MINECRAFT_DAYS =
+            new MinecraftTemporalUnit("HalfMinecraftDays", MINECRAFT_TICKS.getDuration().multipliedBy(12000));
+
+    /**
+     * A {@link MinecraftTemporalUnit} which represents a minecraft day.
+     */
+    public static final MinecraftTemporalUnit MINECRAFT_WEEKS =
```

Comment:
Is there a concept of weeks in minecraft?

Reply:
Not that I know of. A moon cycle is 8 minecraft days, would that be worth adding?

Reply:
That sounds like an implementation detail and I'm not sure whether someone uses that large duration units with a reference to minecraft moon cycles. (The vanilla time is not stable enough enough, for a moonthly message in the chat)